File: 350-IntersectionOfTwoArraysIi/350-IntersectionOfTwoArraysIi.java
// Last updated: 6/25/2026, 8:37:38 AM
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Map<Integer, Integer> freqMap = new HashMap<>();
        int[] n = nums1.length>nums2.length? nums1:nums2;
        for(int i = 0; i < n.length; i++){
            freqMap.put(n[i], freqMap.getOrDefault(n[i], 0) + 1);
        }
        
        n = nums1.length<=nums2.length? nums1:nums2;
        List<Integer> numsIntersect = new ArrayList<>();
        for(int i = 0; i < n.length; i++){
            if(freqMap.getOrDefault(n[i], 0) != 0){
                numsIntersect.add(n[i]);
                freqMap.put(n[i], freqMap.get(n[i]) - 1);
            }
        }
        int[] resArr = new int[numsIntersect.size()];
        int count = 0;
        for(Integer i: numsIntersect){
            resArr[count++] = i.intValue();
        }
        return resArr;
    }
}

File: 350-IntersectionOfTwoArraysIi/350-IntersectionOfTwoArraysIi.py
# Last updated: 6/25/2026, 8:37:44 AM
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1freq = {}
        result = []
        for i in nums1:
            if(i in nums1freq) == False:
                nums1freq[i] = 1
            else:
                nums1freq[i] += 1
        for i in nums2:
            if(i in nums1freq) == True and nums1freq[i] > 0:
                result.append(i)
                nums1freq[i] -= 1
        
        return result
            
        